public/editor/scripts/editor/js/fc/publisher.js

```diff
@@ -122,13 +123,21 @@ define(function(require) {
     });
   };
 
-  Publisher.prototype.publish = function() {
+  Publisher.prototype.publish = function(bramble) {
     var publisher = this;
     var dialog = publisher.dialog;
 
     if (publisher.publishing) {
       return;
     }
+
+    if(!bramble.hasIndexFile()){
+        dialog.buttons.index_message.css("display","block");
+        return;
+    } else {
```

Comment:
You never need/want an `else` after you have a return.  It makes you indent code that doesn't actually need to be.  Always do:

```js
if(...) {
  ...
  return;
}

// no else, keep going here with "else" case
```